language/diem-framework/modules/CRSN.move

```diff
@@ -24,8 +26,23 @@ module DiemFramework::CRSN {
         min_nonce: u64,
         size: u64,
         slots: BitVector,
+        force_shift_events: EventHandle<ForceShiftEvent>,
     }
 
+    /// Whenever a force shift is performed a `ForceShiftEvent` is emitted
+    struct ForceShiftEvent has drop, store {
```

Comment:
I guess we also need a rust-equivalent definition when running through the event stream?

Reply:
nit: add a comment on what the `ForceShiftEvent` is needed for, like "Used to prove the absence of a user transaction at a particular CRSN slot" or something